features-sdk/step_definitions/controllers-steps.js

```diff
@@ -165,3 +166,41 @@ Then('I got an error with id {string}', function (id) {
 
   assert(this.props.error.id === id, `Expected error to have id "${id}", but got "${this.props.error.id}"`);
 });
+
+When('I call the deprecated publicApi method', async function () {
+  if (process.env.KUZZLE_PROTOCOL !== 'http') {
```

Comment:
You can use the `@realtime` annotation at the top of your feature definition to run it only with websocket and mqtt ([see here](https://github.com/kuzzleio/kuzzle/blob/master/features-sdk/Application.feature#L21))

Reply:
I only want it to run when protocol is "http"

Reply:
But i will create an @http then like the @development

Reply:
Ok but I don't think you need to create any extra steps since it's already handled in cucumber profiles
See:
 - https://github.com/kuzzleio/kuzzle/blob/master/package.json#L23
 - https://github.com/kuzzleio/kuzzle/blob/master/cucumber.js#L28